src/renderer/viz/expressions/classification/GlobalMeanStandardDev.js

```diff
@@ -0,0 +1,103 @@
+ *
+ * @example <caption>Use global mean-standard deviation to define a color ramp.</caption>
+ * const s = carto.expressions;
+ * const viz = new carto.Viz({
+ *   color: s.ramp(s.globalMeanStandardDev(s.prop('density'), 5), s.palettes.TEALROSE)
+ * });
+ *
+ * @example <caption>Use global mean-standard deviation to define a color ramp. (String)</caption>
+ * const viz = new carto.Viz(`
+ *   color: ramp(globalMeanStandardDev($density, 5), TEALROSE)
+ * `);
+ *
+ * @example <caption>Use global custom mean-standard deviation to define a color ramp.</caption>
+ * const s = carto.expressions;
+ * const viz = new carto.Viz({
+ *   color: s.ramp(s.globalMeanStandardDev(s.prop('density'), 7, 0.5), s.palettes.TEALROSE)
+ * });
+ *
+ * @example <caption>Use global custom mean-standard deviation to define a color ramp. (String)</caption>
+ * const viz = new carto.Viz(`
+ *   color: ramp(globalMeanStandardDev($density, 7, 0.5), TEALROSE)
+ * `);
+ *
+ * @memberof carto.expressions
+ * @name globalMeanStandardDev
+ * @function
+ * @api
+ */
+export default class GlobalMeanStandardDev extends Classifier {
+    constructor (input, buckets, classSize = 1.0) {
+        checkMaxArguments(arguments, 3, 'globalMeanStandardDev');
+        checkInstance('globalMeanStandardDev', 'input', 0, Property, input && (input.property || input));
+        checkNumber('globalMeanStandardDev', 'buckets', 1, buckets);
+        checkNumber('globalMeanStandardDev', 'classSize', 2, classSize);
+
+        super({ input }, buckets);
+        this._classSize = classSize;
+        this._sample = [];
+    }
+
+    _bindMetadata (metadata) {
+        super._bindMetadata(metadata);
+        checkExpression('globalMeanStandardDev', 'input', 0, this.input);
+        checkType('globalMeanStandardDev', 'input', 0, 'number', this.input);
+
+        const sample = metadata.sample.map(s => s[this.input.name]);
+        const avg = average(sample);
+        const standardDev = standardDeviation(sample);
+
+        const breaks = calculateBreakpoints(avg, standardDev, this.buckets, this._classSize);
+        this.breakpoints.forEach((breakpoint, index) => {
+            breakpoint.expr = breaks[index];
+        });
+    }
+}
+
+/**
+ * Calculate breakpoints according to mean-standard deviation process
+ *
+ * @export
+ * @param {Number} avg - average
+ * @param {Number} stDev - standard deviation
+ * @param {Number} buckets - number of buckets
+ * @param {Number} classSize - in standard deviation units (usually 1.0, 0.5, 0.25...)
+ * @returns {Number[]}
+ */
+export function calculateBreakpoints (avg, stDev, buckets, classSize) {
+    let breaks;
+    let over = [];
+    let under = [];
+    const isEven = buckets % 2 === 0;
+    let factor = isEven ? 0.0 : 1.0; // if odd, central class is double sized
+    do {
+        const step = factor * (stDev * classSize);
+        over.push(avg + step);
+        under.push(avg - step);
+        breaks = [...new Set(over.concat(under))];
```

Comment:
I don't understand this line. I guess that using a set is to remove duplicates, but I don't see why elements of `over` and elements of `under` may be equal

Reply:
Ok, now I see that `step` will be `0` in the first iteration when `isEven`

Reply:
That's right, that is the only corner case